### DISCUSSION TITLE
[RFC] Expose the `Item` in the `Space` component

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#42702] [Ideas]
## Summary

Explicitly give developers access to the internal Space [Item](https://github.com/ant-design/ant-design/blob/master/components/space/Item.tsx) component that is created by the `Space` component for each of its children.

## Motivation

The `<Space />` [component](https://ant.design/components/space) is a great layout component with very convenient functionality, but sometimes, finer-tuned behavior and styling are desired for the Children component. For example, updating the flex size properties of individual children (currently not possible since each is wrapped by an inaccessible `<div class="ant-space-item">...</div>` component or adjusting the justify properties of the space items and group.

I am proposing to expose the space item using the `<Space.Item>` syntax to have more control over the styling of a child while keeping the affordances provided by `<Space />`.

### Space.Item

(Taken from the existing code)

Property | Description | Type | Default
--- | --- | --- | ---
className | Semantic DOM class  | string | -
children | The wrapped item component | ReactNode | -
index | Index of the items  | number | -
direction | Flex direction | `horizontal` \| `vertical` | `horizontal`
marginDirection | Side where to add margin | `marginLeft` \| `marginRight` | -
split | If defined, the item is split in 2, and this component is rendered next to the child   | ReactNode | -
wrap | Add bottom padding | boolean | -
style | Semantic DOM style | CSSProperties | -

## Basic Example

```tsx
<Space style={{  display: 'flex', justifyContent: 'space-between' }}>
  <Typography.Text>This is some test.</Typography.Text> // <-- current behavior
  <Space.Item style={{ flex: 1}} className="my-class" > // <-- new behavior, can add className/styles to `<div class="ant-space-item">...</div>` 
    <Button>Click me</Button>
  </ Space.Item>
</ Space>
```
Result:
```html
<div class="ant-space " style="justify-content: space-between; display: flex;">
  <div class="ant-space-item">
    <span class="ant-typography">This is some test</span>
  </div>
  <div class="ant-space-item my-class" style="flex: 1;">
    <button style="flex: 1 1 0%;" type="button" class="ant-btn">
    ...
    </button>
  </div>
</div>
```

## How we achieved it

- [ ] Export/expose the Item subcomponent

**Comment by Wxh16144** [comment]
Is this a duplicate of the `<Grid />` [component](https://ant.design/components/grid)?

**Reply by bombillazo** [reply]
In functionality, one could achieve the same result, but in usability, it gives more options to have quick control with the Space component.

**Comment by bombillazo** [comment]
Any other feedback from the core team? @MadCcc @kiner-tang @afc163 @li-jia-nan @zombieJ

**Comment by samiralibabic** [comment]
One more vote 👍

Reason: I'm unable, without trickery, to make input take the full width of the parent container because of this. Look at this example:
![carbon (1)](https://github.com/ant-design/ant-design/assets/17047213/3804b35d-5809-4bfa-af98-449c9dfbd1b0)

**Reply by bombillazo** [reply]
Just recently hit this use case when using the `Input` component as well , the exact same setup (input + button)

**Comment by bombillazo** [comment]
@afc163 @zombieJ sorry for tagging but is this feature possible? We've found frequent needs for this to be able to control better the styling of Space children elements ad-hoc without needing CSS
